apps/api/v2/src/modules/organizations/attributes/options/organization-attribute-options.repository.ts

```diff
@@ -90,6 +92,78 @@ export class OrganizationAttributeOptionRepository {
     return options;
   }
 
+  async getOrganizationAttributeAssignedOptions({
+    attributeId,
+    attributeSlug,
+    filters,
+    organizationId,
+    skip,
+    take,
+  }: GetOrganizationAttributeAssignedOptionsProp) {
+    const options = await this.dbRead.prisma.attributeOption.findMany({
+      where: {
+        attribute: {
+          ...(attributeId && { id: attributeId }),
+          ...(attributeSlug && { slug: attributeSlug }),
+          teamId: organizationId,
+        },
+        assignedUsers: {
+          some: {}, // empty {} statement checks if option is assigned to at least one user
+          ...(filters?.teamIds && {
+            some: { member: { user: { teams: { some: { teamId: { in: filters.teamIds } } } } } },
+          }),
+        },
+      },
+      include: { assignedUsers: { include: { member: true } } },
+      skip,
+      take,
+    });
+
+    // only return options that are assigned to users alongside assignedOptionIds filter
+    if (filters?.assignedOptionIds) {
+      const filteredAssignedOptions = await this.dbRead.prisma.attributeOption.findMany({
+        where: {
+          attribute: { teamId: organizationId },
+          assignedUsers: {
+            every: {
+              attributeOptionId: { in: filters?.assignedOptionIds },
+            },
+            ...(filters?.teamIds && { some: { member: { teamId: { in: filters?.teamIds } } } }),
+          },
+        },
+        include: { assignedUsers: { include: { member: true } } },
+      });
+
+      if (!filteredAssignedOptions?.length) {
+        throw new NotFoundException("Options provided in assignedOptionIds are not assigned to anyone.");
```

Comment:
There could be other reasons for this as well mainly teamIds could be other reason.